src/readme.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 Microsoft Corporation
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+import * as commonmark from 'commonmark'
+import { parse } from '@ts-common/commonmark-to-markdown'
+import { getTagsToSettingsMapping, getCodeBlocksAndHeadings } from '@azure/openapi-markdown'
+import * as fs from 'fs'
+import { safeLoad } from './utils'
+
+/**
+ * walks a markdown tree until the callback provided returns true for a node
+ */
+export const walkToNode = (
+  walker: commonmark.NodeWalker,
+  cb: (node: commonmark.Node) => boolean,
+): commonmark.Node | undefined => {
+  let event = walker.next()
+
+  while (event) {
+    const curNode = event.node
+    if (cb(curNode)) {
+      return curNode
+    }
+    event = walker.next()
+  }
+  return undefined
+}
+
+export const sortByApiVersion = (versions: string[]): string[] => {
+  // sort by api version. Format: YYYY-MM-DD
+  const supportedRegex = [/(\d{4})-(\d{2})-(\d{2})/, /(\d{4})_(\d{2})_(\d{2})/]
+
+  for (const regex of supportedRegex) {
+    const filterVersion = versions.filter(v => regex.test(v))
+    if (filterVersion.length > 0) {
+      return filterVersion.sort((a, b) => {
+        const aMatch = a.match(regex)!
+        const bMatch = b.match(regex)!
+        const aDate = new Date(+aMatch[1], +aMatch[2], +aMatch[3])
```

Comment:
instead of having multiple matching groups, consider using `/(\d{4}-\d{2}-\d{2})/` to get entire match, and then create date by `new Date(match)`

Reply:
No.Currently, we need to support YYYY_MM_DD format. We cannot use new Date() with YYYY_MM_DD format.

Reply:
For `YYYY_MM_DD`, we can replace `_` with `-`

Reply:
I don't want to remove such code changes. In Current solution if we remove the support we just remove it from support regex list. It's clear.

Reply:
Sure. Once remove it from regex list, remember to simplify the matching pattern as well